### PR TITLE
[5.1] whereIn can now convert a collection to an array

### PR DESCRIPTION
**Pull request by jbrooksuk**
This prevents #9013 from happening.

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -748,6 +748,8 @@ public function whereIn($column, $values, $boolean = 'and', $not = false)
         // execute those Closures, then we can re-construct the entire sub-selects.
         if ($values instanceof Closure) {
             return $this->whereInSub($column, $values, $boolean, $not);
+        } elseif ($values instanceof Collection) {
+            return $this->whereIn($column, $values->toArray(), $boolean, $not);
```
Comment (lukasgeiter) [inline src/Illuminate/Database/Query/Builder.php]: Why not just `$values = $values->toArray()` ?

Comment (jbrooksuk) [inline src/Illuminate/Database/Query/Builder.php]: My thought was that if something happens before the condition, the method needs to be re-called. I should probably change this for now though.

**Comment by crynobone** [comment]
Shouldn't this go to `5.1` branch?

**Comment by lukasgeiter** [comment]
Yes, definitely the 5.1 branch!

**Comment by jbrooksuk** [comment]
Whoops, my bad.

**Comment by jbrooksuk** [comment]
I'll re-open this with the right branch!

**Comment by jbrooksuk** [comment]
#9015

**Pull request closed without merging**